invoice_discount/model/invoice.py

```diff
@@ -23,133 +23,65 @@
-        invoice lines.
-        @param self: The object pointer.
-        @param cr: A database cursor
-        @param uid: ID of the user currently logged in
-        @param ids: list of ids for which name should be read
-        @param field_name: field that call the method
-        @param arg: Extra arguments
-        @param context: A standard dictionary
-        @return : Dict with values
-        """
-        context = context or {}
-        total = 0.0
-        res = {}
-        for inv in self.browse(cr, uid, ids, context=context):
-            for line in inv.invoice_line:
-                total += line.subtotal_wo_discount
-            res[inv.id] = total
-        return res
-
-    def _get_discount(self, cr, uid, ids, args, field_name, context=None):
-        """Method to get the amount total of discount in the invoice lines.
-        @param self: The object pointer.
-        @param cr: A database cursor
-        @param uid: ID of the user currently logged in
-        @param ids: list of ids for which name should be read
-        @param field_name: field that call the method
-        @param arg: Extra arguments
-        @param context: A standard dictionary
-        @return : Dict with values
-        """
-        context = context or {}
-        total = 0.0
-        res = {}
-        for inv in self.browse(cr, uid, ids, context=context):
-            for line in inv.invoice_line:
-                total += line.discount_amount
-            res[inv.id] = total
-        return res
+    subtotal_wo_discount = fields.Float(
+        compute='_compute_subtotal_wo_disc', string='SubTotal w/o Discount',
+        store=True,
+        help='Amount without apply the discount of the lines of the invoice.')
+    discount_amount = fields.Float(
+        compute='_compute_subtotal_wo_disc',  string='Discount', store=True,
+        help='Total of discount apply in each line of the invoice.')
 
-    _columns = {
-        'subtotal_wo_discount': fields.function(_get_subtotal_without_discount,
-                                                string='SubTotal w/o Discount',
-                                                store=True, type='float',
-                                                help='Amount without apply the\
-                                                 discount of the lines of the \
-                                                invoice.'),
-        'discount_amount': fields.function(_get_discount, string='Discount',
-                                           store=False, type='float',
-                                           help='Total of discount apply in \
-                                           each line of the invoice.'),
-    }
+    @api.depends('invoice_line.subtotal_wo_discount',
+                 'invoice_line.discount_amount')
+    def _compute_subtotal_wo_disc(self):
+        # TODO: Use read_group method here
+        for invoice in self:
```

Comment:
Hello @moylop260 
I think the same way that you do. read_group is the fastest solution.

I did a test in a copy of apex using two similar invoice_line fields
PSQL:
![desktop screenshot 1](https://user-images.githubusercontent.com/5271213/29133486-401b1012-7cf9-11e7-8b6c-8f4ea188664b.png)

read_group with odoorpc
![desktop screenshot 2](https://user-images.githubusercontent.com/5271213/29134756-634a8fe6-7cfd-11e7-90d1-0ec6dca849c3.png)
In our case would be something like this:
```python
line = self.env['account.invoice.line']
for invoice in self:
    results = line.read_group([('invoice_id', '=', invoice.id)], ['discount_amount', 'subtotal_wo_discount', 'invoice_id'], ['invoice_id'])
    invoice.subtotal_wo_discount = results and results[0].get('subtotal_wo_discount') or 0
    invoice.discount_amount = results and results[0].get('discount_amount') or 0
```
There is an important question. 
Are these fields stored in database?
 The read_group method will be useful, only if these fields are stored in database, if one of them isn't we will have to use panda as @hbto has said before